### PR TITLE
Matrix exports

### PR DESCRIPTION
**Pull request by pedrocamargo**
I'm separating matrix exports from the deprecation corrections, as I have NO idea of what's going on there

### CODE REVIEW

**Comment by pedrocamargo** [comment]
@Jake-Moss or @jamiecook , can you take a look at what the hell is going on here? We can chat a bit about it when we are both on deck as well.  I am COMPLETELY at loss

File: aequilibrae/matrix/aequilibrae_matrix.py [hunk 1]
```diff
@@ -748,18 +752,14 @@ def export(self, output_name: str, cores: List[str] = None):
             >>> mat2.cores
             2
         """
-
-        if self.__omx:
-            raise NotImplementedError("This operation does not make sense for OMX matrices")
-
         fname, file_extension = os.path.splitext(output_name.upper())
```
Comment (jamiecook) [inline aequilibrae/matrix/aequilibrae_matrix.py:755]:
```suggestion
        fname, file_extension = os.path.splitext(output_name.upper())
        file_extension = file_extension.upper()
```
?

**Review by jamiecook — APPROVED** [review]

**Comment by jamiecook** [comment]
You are referring to the failing test for windows 3.7 right?

**Comment by jamiecook** [comment]
logs keep on getting deleted - so i'm uploading here:
[windows_37.log](https://github.com/AequilibraE/aequilibrae/files/12139897/windows_37.log)

**Comment by jamiecook** [comment]
Key lines seem to be here: 

```
2023-07-23T20:21:58.6386342Z   File "C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\site-packages\_pytest\fixtures.py", line 911, in _teardown_yield_fixture
2023-07-23T20:21:58.6386914Z     next(it)
2023-07-23T20:21:58.6387633Z   File "C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\site-packages\_pytest\tmpdir.py", line 283, in tmp_path
2023-07-23T20:21:58.6388280Z     result_dict = request.node.stash[tmppath_result_key]
2023-07-23T20:21:58.6389082Z   File "C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\site-packages\_pytest\stash.py", line 80, in __getitem__
2023-07-23T20:21:58.6389888Z     return cast(T, self._storage[key])
2023-07-23T20:21:58.6390407Z KeyError: <_pytest.stash.StashKey object at 0x0000022247C0B490>
```

Something is going wrong in the pytest fixture cleanup / teardown .... it's looking for a key in a "stash" that it's expecting to be there but for some reason isn't... perhaps a multithreading bug? I'm guessing it's in the omx tests... seeing as that is what is being changed here... but it could just be a pytest bug... 

I'm going to open a clean PR with a doco change to see if it triggers the same

**Comment by jamiecook** [comment]
Looks like a failure on 3.11 is causing other tests to get cancelled. 

image

**Pull request closed without merging**